core/trino-main/src/main/java/io/trino/execution/scheduler/EventDrivenFaultTolerantQueryScheduler.java

```diff
@@ -484,6 +494,8 @@ private static class Scheduler
             implements EventListener
     {
         private static final int EVENT_BUFFER_CAPACITY = 100;
+        private static final int RUNTIME_ADAPTIVE_PLAN_NODE_ID_THRESHOLD = 1_000_000_000;
+        private static final int RUNTIME_ADAPTIVE_PLAN_FRAGMENT_ID_THRESHOLD = 1_000_000_000;
```

Comment:
this is extremly hacky - not sure why we need special tagging of plan fragments. But if have to we probably can do it externally via some map/set. Or maybe we do not need any explicit tagging but we can just look at plannod/plan fragment properties.

Reply:
Done